webxr/resources/webxr_util.js

```diff
@@ -10,21 +10,24 @@
 // Debugging message helper, by default does nothing. Implementations can
 // override this.
 var xr_debug = function(name, msg) {}
+var isChromiumBased = 'MojoInterfaceInterceptor' in self;
+var isWebKitBased = 'internals' in self && 'xrTest' in internals;
 
 function xr_promise_test(name, func, properties) {
   promise_test(async (t) => {
     // Perform any required test setup:
     xr_debug(name, 'setup');
 
-    if (window.XRTest === undefined) {
-      // Chrome setup
-      await loadChromiumResources;
-      xr_debug = navigator.xr.test.Debug;
-    }
-
-    if (self.internals && internals.xrTest && navigator.xr) {
-      // WebKit setup
-      await setupWebKitWebXRTestAPI;
+    if (!navigator.xr.test) {
+        if (isChromiumBased) {
+          // Chrome setup
+          loadChromiumResources().then(() => {
```

Comment:
Please leave this as:
```
await loadChromiumResources();
xr_debug = navigator.xr.test.Debug;
```
as otherwise we would no longer be waiting on chromium setup.